system-parachains/bridge-hubs/bridge-hub-polkadot/src/bridge_to_ethereum_config.rs

```diff
@@ -227,6 +227,141 @@ pub mod benchmark_helpers {
 	}
 }
 
+pub mod migration {
+	#[cfg(feature = "try-runtime")]
+	use crate::{vec, Vec};
+	use frame_support::weights::Weight;
+	#[cfg(feature = "try-runtime")]
+	use hex_literal::hex;
+	use snowbridge_core::TokenId;
+	use sp_core::Get;
+	#[cfg(feature = "try-runtime")]
+	use sp_runtime::TryRuntimeError;
+	use xcm::prelude::*;
+
+	pub struct RegisteredPNA {
+		pub location: Location,
+		pub token_id: TokenId,
+	}
+
+	/// Migrate ForeignToNativeId from XCMv4 to XCMv5.
+	pub struct MigrateLocationsToXcmV5<T: snowbridge_pallet_system::Config>(
+		core::marker::PhantomData<T>,
+	);
+
+	impl<T: snowbridge_pallet_system::Config> frame_support::traits::OnRuntimeUpgrade
+		for MigrateLocationsToXcmV5<T>
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let mut weight = T::DbWeight::get().reads_writes(1, 1);
+
+			let translate = |pre: xcm::v4::Location| -> Option<xcm::v5::Location> {
+				weight.saturating_accrue(T::DbWeight::get().reads_writes(1, 2));
+				let location = xcm::v5::Location::try_from(pre.clone()).expect("valid location");
```

Comment:
@yrong I think that using `.expect(` here is not the best thing:

https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/support/src/traits/hooks.rs#L473-L507
```
/// Similar to [`Hooks::on_initialize`], any code in this block is mandatory and MUST execute.
/// It is strongly recommended to dry-run the execution of these hooks using
/// [try-runtime-cli](https://github.com/paritytech/try-runtime-cli) to ensure they will not
/// produce and overweight block which can brick your chain. Use with care!
...
fn on_runtime_upgrade() -> Weight {
```